datafusion-cli/src/exec.rs

```diff
@@ -88,7 +88,8 @@ pub async fn exec_from_lines(
     }
 
     // run the left over query if the last statement doesn't contain ‘;’
-    if !query.is_empty() {
+    // ignore if it only consists of '\n'
+    if query.contains(|c| c != '\n') {
```

Comment:
maybe we could also check for spaces?

```suggestion
    if query.contains(|c| c != '\n' && c != ' ') {
```

Reply:
@alamb Thank you for looking into this PR!
I don't think we need to checn `' '` here because lines are [trimmed](https://github.com/apache/arrow-datafusion/pull/7076/files#diff-00370e7628c93a0c9da5107fa0d065240ff09d0425d9559b8662fb0b2e10feebL72).
Actually, `' '` and any other whitespaces like `U+3000` are ignored even though they are at the end of a file.